### PR TITLE
Add journald integration test

### PR DESCRIPTION
**Pull request by jorgebay**
Ref: LOG-8374

### CODE REVIEW

File: bin/tests/cli.rs [hunk 1]
```diff
@@ -391,3 +396,36 @@ fn test_directory_symlinks_delete() {
...
+        &mut stderr_reader,
+    );
+
+    for i in 0..10 {
+        journal::print(1, format!("Sample alert {}", i).as_str());
+        journal::print(6, format!("Sample info {}", i).as_str());
+    }
+
+    common::wait_for_event("sending journald line", &mut stderr_reader);
+    common::wait_for_event("sending http request", &mut stderr_reader);
```
Comment (jorgebay) [inline bin/tests/cli.rs:427]: Not very elegant...

Once we have the mock ingester as a dev dependency we can verify what was received.

File: bin/tests/common.rs [hunk 2]
```diff
@@ -172,7 +184,7 @@ pub fn create_dirs<P: AsRef<Path>>(dirs: &[P]) {
 
 pub fn open_files_include(id: u32, file: &PathBuf) -> Option<String> {
     let child = Command::new("lsof")
-        .args(&["-p", &id.to_string()])
+        .args(&["-l", "-p", &id.to_string()])
```
Comment (jorgebay) [inline bin/tests/common.rs:187]: lsof was showing in the output "no pwd entry for UID" which was irritating, I've fixed it here even though it's unrelated to this PR.

File: docker/journald/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,4 @@
 FROM docker.io/logdna/build-images:rust-buster-stable
-RUN apt update -y && apt install -y systemd lsof procps
```
Comment (jorgebay) [inline docker/journald/Dockerfile:2]: We should clean the image cache on our local dev env as a result of this.

File: bin/tests/common.rs [hunk 4]
```diff
@@ -184,32 +185,43 @@ pub fn spawn_agent(settings: AgentSettings) -> Child {
...
 
+    if let Some(journald_dirs) = settings.journald_dirs {
+        agent.env("LOGDNA_JOURNALD_PATHS", journald_dirs);
+    }
+
     agent.spawn().expect("Failed to start agent")
 }
 
+/// Blocks until a certain event referencing a file name is logged by the agent
+pub fn wait_for_file_event(event: &str, file_path: &PathBuf, reader: &mut dyn BufRead) -> String {
```
Comment (c-nixon) [inline bin/tests/common.rs:196]: 💯

**Review by c-nixon — APPROVED** [review]

File: bin/tests/common.rs (hunk 4 above)
Comment (c-nixon) [reply bin/tests/common.rs:196]: The &PathBuf could probably be a Path, but there's very little difference

Comment (jorgebay) [reply bin/tests/common.rs]: Changed it!

**Pull request merged**